File: .claude/skills/skill-creator/templates/example-script.py
#!/usr/bin/env python3
"""
Example Python Script Template for Skills

This template demonstrates best practices for writing Python scripts
that can be included in Custom Skills.
"""

import argparse
import json
import sys
from pathlib import Path
from typing import Dict, Any, List


def parse_arguments() -> argparse.Namespace:
    """
    Parse command line arguments.

    Returns:
        Parsed command line arguments
    """
    parser = argparse.ArgumentParser(
        description='Example script for Custom Skill'
    )
    parser.add_argument(
        '--input',
        type=str,
        required=True,
        help='Input file path'
    )
    parser.add_argument(
        '--output',
        type=str,
        required=True,
        help='Output file path'
    )
    parser.add_argument(
        '--verbose',
        action='store_true',
        help='Enable verbose output'
    )
    return parser.parse_args()


def validate_input(input_path: str) -> bool:
    """
    Validate that the input file exists and is readable.

    Args:
        input_path: Path to input file

    Returns:
        True if valid, False otherwise
    """
    path = Path(input_path)
    if not path.exists():
        print(f"Error: Input file '{input_path}' does not exist", file=sys.stderr)
        return False
    if not path.is_file():
        print(f"Error: '{input_path}' is not a file", file=sys.stderr)
        return False
    return True


def process_data(data: Dict[str, Any]) -> Dict[str, Any]:
    """
    Process the input data and return results.

    Args:
        data: Input data dictionary

    Returns:
        Processed data dictionary
    """
    # Add your processing logic here
    result = {
        'status': 'success',
        'input_keys': list(data.keys()),
        'processed': True
    }
    return result


def main():
    """Main execution function."""
    args = parse_arguments()

    # Validate input
    if not validate_input(args.input):
        sys.exit(1)

    try:
        # Read input file
        with open(args.input, 'r') as f:
            input_data = json.load(f)

        if args.verbose:
            print(f"Processing {args.input}...")

        # Process data
        result = process_data(input_data)

        # Write output
        with open(args.output, 'w') as f:
            json.dump(result, f, indent=2)

        if args.verbose:
            print(f"Results written to {args.output}")
            print(f"Status: {result['status']}")

        return 0

    except json.JSONDecodeError as e:
        print(f"Error: Invalid JSON in input file: {e}", file=sys.stderr)
        return 1
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        return 1


if __name__ == '__main__':
    sys.exit(main())

File: plugin-repo/skills/weaver-create/templates/example-script.js
#!/usr/bin/env node
/**
 * Example JavaScript Script Template for Skills
 *
 * This template demonstrates best practices for writing Node.js scripts
 * that can be included in Custom Skills.
 */

const fs = require('fs').promises;
const path = require('path');

/**
 * Parse command line arguments
 * @returns {Object} Parsed arguments
 */
function parseArguments() {
    const args = process.argv.slice(2);
    const parsed = {
        input: null,
        output: null,
        verbose: false
    };

    for (let i = 0; i < args.length; i++) {
        switch (args[i]) {
            case '--input':
                parsed.input = args[++i];
                break;
            case '--output':
                parsed.output = args[++i];
                break;
            case '--verbose':
                parsed.verbose = true;
                break;
            case '--help':
                printHelp();
                process.exit(0);
            default:
                console.error(`Unknown argument: ${args[i]}`);
                printHelp();
                process.exit(1);
        }
    }

    if (!parsed.input || !parsed.output) {
        console.error('Error: --input and --output arguments are required');
        printHelp();
        process.exit(1);
    }

    return parsed;
}

/**
 * Print help message
 */
function printHelp() {
    console.log(`
Usage: node example-script.js [options]

Options:
  --input <path>    Input file path (required)
  --output <path>   Output file path (required)
  --verbose         Enable verbose output
  --help            Show this help message
    `.trim());
}

/**
 * Validate that input file exists
 * @param {string} inputPath - Path to input file
 * @returns {Promise<boolean>} True if valid
 */
async function validateInput(inputPath) {
    try {
        const stats = await fs.stat(inputPath);
        if (!stats.isFile()) {
            console.error(`Error: '${inputPath}' is not a file`);
            return false;
        }
        return true;
    } catch (error) {
        console.error(`Error: Input file '${inputPath}' does not exist`);
        return false;
    }
}

/**
 * Process the input data
 * @param {Object} data - Input data
 * @returns {Object} Processed data
 */
function processData(data) {
    // Add your processing logic here
    return {
        status: 'success',
        inputKeys: Object.keys(data),
        processed: true,
        timestamp: new Date().toISOString()
    };
}

/**
 * Main execution function
 */
async function main() {
    const args = parseArguments();

    // Validate input
    if (!await validateInput(args.input)) {
        process.exit(1);
    }

    try {
        // Read input file
        const inputContent = await fs.readFile(args.input, 'utf8');
        const inputData = JSON.parse(inputContent);

        if (args.verbose) {
            console.log(`Processing ${args.input}...`);
        }

        // Process data
        const result = processData(inputData);

        // Write output
        await fs.writeFile(
            args.output,
            JSON.stringify(result, null, 2),
            'utf8'
        );

        if (args.verbose) {
            console.log(`Results written to ${args.output}`);
            console.log(`Status: ${result.status}`);
        }

        process.exit(0);

    } catch (error) {
        if (error instanceof SyntaxError) {
            console.error(`Error: Invalid JSON in input file: ${error.message}`);
        } else {
            console.error(`Error: ${error.message}`);
        }
        process.exit(1);
    }
}

// Run main function
main().catch(error => {
    console.error(`Unexpected error: ${error.message}`);
    process.exit(1);
});
